src/containers/Signup.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useAlert } from 'react-alert';
+import { Link, useNavigate } from 'react-router-dom';
+import { setErrorSigninAction } from '../actions';
+
+const Signup = () => {
+  const [state, setState] = useState({ name: '', user: '', password: '' });
```

Comment:
The more proper way to use React's useState hook is as follows:

const [name, setName] = useState('');
const [user, setUser] = useState('');
const [password, usePassword] = useState('');

Setting all of those values in one state is working against why React created useState in the first place - to more efficiently manage local state.